### DISCUSSION TITLE
Server configuration on Fully managed VPS on TMD Hosting

### DISCUSSION THREAD

**Discussion by naas1172** [YetiForceCompany/YetiForceCRM#15345] [General]
Hello everyone.
We have a fully managed VPS Hosting in TMD Hosting.
Yetiforce is installed. The support team has helped me a lot, but there are still some issues. 
Yetiforce is up to date: YetiForce CRM 6.2 Service Pack v4.
Please check the images here below. 
In the last email Support wrote me: "Unfortunately, your application does not detect the rest of the changes we have made and the warnings are still present. However, these should not be preventing the application from working properly."
Do you think they should be able to fix the remaining issue? Is it safe to use the CRM with these issues?

Thank you!
Alessandro

![system_headers_YetiForce_6-2_Service-Pack-v4](https://user-images.githubusercontent.com/92877296/138132222-a48b9dd9-fa69-4c24-a4ff-ec33bc089fc6.jpg)
![system_path-verification_YetiForce_6-2_Service-Pack-v4](https://user-images.githubusercontent.com/92877296/138132226-141e7876-96e5-48df-8f80-2be1d09587fd.jpg)
![system_performance-verification_YetiForce_6-2_Service-Pack-v4](https://user-images.githubusercontent.com/92877296/138132229-57a4e095-8334-493a-b18e-a385f830222a.jpg)
![system_security_YetiForce_6-2_Service-Pack-v4](https://user-images.githubusercontent.com/92877296/138132235-f9c7a129-d5a3-483e-a2e3-b4432706c36e.jpg)

**Comment by vovpff** [comment]
You need to run cron once to check cron (cli) column parameters

**Reply by naas1172** [reply]
Hi there. Cron is already set, it runs every 5 mins. No changes whatsoever.

**Reply by apcloic** [reply]
@vovpff thanks a lot for the tips, I was fighting with an empty cron (CLI) column on a new test environment :)

**Comment by naas1172** [comment]
Is it safe to use the CRM with these issues?

**Reply by mariuszkrzaczkowski** [reply]
attach full confreport report
![image](https://user-images.githubusercontent.com/10198654/142369414-1e8c1635-c0e4-4155-94ea-aaa6203b6b43.png)

**Comment by naas1172** [comment]
Hello Mariusz.
Thank you for your answer.
Please find here below the full confreport.

Ciao
Alessandro

![ConfReport](https://user-images.githubusercontent.com/92877296/142646611-c79b5bcb-e2f4-4137-a4bf-8216cdc5bcd2.png)

**Reply by mariuszkrzaczkowski** [reply]
This needs to be corrected:
1. different php versions
![image](https://user-images.githubusercontent.com/10198654/142818747-61fae857-aecc-442c-8269-5b4a8bb64cae.png)
2. the domain should point to the public_html directory https://github.com/YetiForceCompany/YetiForceCRM/blob/developer/tests/setup/apache/httpd-vhosts.conf#L6
![image](https://user-images.githubusercontent.com/10198654/142818862-74855cb1-bf75-4e93-bb3c-1985eb5e70b0.png)
3. 
![image](https://user-images.githubusercontent.com/10198654/142819157-eb12fc1e-1d1f-4362-9fa1-6d17157f7cc4.png)
4. https://yetiforce.com/pl/component/zoo/item/web-server-requirements#WebServer
![image](https://user-images.githubusercontent.com/10198654/142819277-3a6ec50e-8ac9-4f77-aac0-2c94148e9203.png)

**Reply by naas1172** [reply]
Hi Marius.
Thank you very much!

Regards
Alessandro

**Reply by naas1172** [reply]
Hi.
The hosting provider has done some modifications, but the issues regarding public_html and realpath are still there. 
This is what they told me.
About the public_html issue: "Your domain name is indeed serving its content from the public_html directory as it is the main domain name associated with the cPanel account. In case this is a path that needs to be set in a configuration file, kindly specify the exact file that this should be modified in."
About the realpath: "We have managed to resolve the opcache related issues, however, the realpath one remains since there is no directive in which to set the parameters as this might be deprecated or not present in the PHP Version set in use."
You can find attached the current configuration report.
![20211126-ConfReport](https://user-images.githubusercontent.com/92877296/143615225-166e177c-6567-40c5-9143-31cb4f0cc800.png)

Anything i can tell them? 
Thank you!

Alessandro

**Reply by naas1172** [reply]
They have managed to align the php versions that were still misaligned.
Now they're both 7.4.26, Apache and Cron.
I'll push them for the other two issues that they haven't yet resolved.
I don't get why it is this difficult.

Alessandro

**Reply by mariuszkrzaczkowski** [reply]
attach the current report

**Reply by naas1172** [reply]
Hello Mariusz.
Here it is, attached.
Regarding the public_html issue, this is what they wrote me:

"Since your server is a fully managed Hosting Solution, this means that it takes advantage of the WHM/cPanel interfaces, each new Domain Name that you add has its' Apache Virtual Hosts created automatically within the Apache Configuration, or in other words, the changes that you wish to be performed from the GitHub documentation that you have shared.
Each Primary Domain Name of a cPanel account loads the content of the "public_html" directory within the account by default, meaning that its' Virtual Host is configured to load files from the following directory, which in your case is: /home/schaefer/public_html

However, the Yetiforce application performs the checks differently and does not expect cPanel to be installed on the server, therefore, it simply returns a false-positive result that something is not enabled, while in reality, it is.

Therefore, there is nothing to worry about, and you should simply ignore this warning, as everything is correctly set up here, and no further changes are required."

Thank you!
Alessandro

![20211129-ConfReport-YF](https://user-images.githubusercontent.com/92877296/143927641-01ec297c-7056-4b78-9282-49472f630d5a.png)

**Reply by mariuszkrzaczkowski** [reply]
> However, the Yetiforce application performs the checks differently and does not expect cPanel to be installed on the server, therefore, it simply returns a false-positive result that something is not enabled, while in reality, it is.

You probably have all the CRM files in the directory: /home/schaefer/public_html

so you have two options:
- move all files to the directory below that is /home/schaefer/
- change the root directory path: "DocumentRoot /home/schaefer/public_html/public_html"

The rest looks OK, only minor details are left

**Reply by naas1172** [reply]
Hello Mariusz.
I've contacted assistance and i've sent them yours suggestions, and this their reply:

> "We've reviewed the suggestion provided and made some modifications to the Apache configuration as is required by YetiForce, however the notification is still prevalent when we check the website.
> 
> We configured the Virtual Host of the website exactly as recommended, but once again - no difference is met.
> 
> With that in mind, we believe the notification is a false-positive, as the configuration is properly set and edited. As it stands, the YetiForce CRM application is operating without any issues, nor are we able to locate any complications on it despite the notification.
> 
> Considering all of this, we would like to assure you that the application is running properly and the required changes in order for it to operate seamlessly have been made."

Should i try Yetiforce like this, or should i ask them other information/modifications?
Thank you very much!

Best,
Alessandro

**Reply by mariuszkrzaczkowski** [reply]
It's hard to say, I don't know the environmental configuration

**Reply by naas1172** [reply]
Thank you for your reply.
Here is the configuration:

The server is using CentOS by default (a custom environment could be deployed to use the OS i wish, upon request).
The MySQL version on your server is 10.2.41.
We have 40GB of SSD Space, 2 CPU Cores, and 2GB DDR4 RAM.

Thank you!

Regards,
Alessandro
